src/request.rs

```diff
@@ -16,11 +19,87 @@ use toml::Table;
 use crate::{
     env::{update_data, HitmanCookieJar},
     extract::extract_variables,
-    prompt::{get_interaction, substitute_interactive},
+    prompt::{get_interaction, prepare_request_interactive},
     util::truncate,
 };
 
-static USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"),);
+#[derive(Clone)]
+pub enum HitmanBody {
+    REST {
+        body: String,
+    },
+    GraphQL {
+        operation: GraphQLOperation,
+        body: String,
+        variables: Option<String>,
+    },
+}
+
+impl Display for HitmanBody {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            HitmanBody::REST { body } => write!(f, "{}", body),
+            HitmanBody::GraphQL {
+                operation: _,
+                body,
+                variables,
+            } => match variables {
+                Some(v) => write!(f, "{}\n{}", body, v),
+                None => write!(f, "{}", body),
+            },
+        }
+    }
+}
+
+impl HitmanBody {
+    pub fn to_body(self) -> String {
+        match self {
+            HitmanBody::REST { body } => body,
+            HitmanBody::GraphQL {
+                operation,
+                body,
+                variables,
+            } => match variables {
+                Some(v) => json!({operation.to_string(): body, "variables": v})
+                    .to_string(),
+                None => json!({operation.to_string(): body}).to_string(),
+            },
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct HitmanRequest {
+    pub headers: HeaderMap,
+    pub url: Url,
+    pub method: Method,
+    pub body: Option<HitmanBody>,
+}
+
+impl Display for HitmanRequest {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "{} {}", self.method.as_str(), self.url.as_str())?;
+        for (key, val) in self.headers.iter() {
```

Comment:
This is also a little bit different than before. `HeaderName` which is the type used in `HeaderMap` for the key is actually turning all of the header-keys to lowercase.

So it is printing the keys for the header in lowercase instead of exactly how it is written in the `http` file. Hope that is okay